### DISCUSSION TITLE
New keybinds for P-51D/TF-51D

### DISCUSSION THREAD

**Discussion by correca** [Munkwolf/dcs-community-keybinds#27] [General]
Hi, these are some keybinds I use with P-51D/TF-51D. Some pairs of commands I propose can be assigned to three-position switches on your game controller. 

		-- Flight Control
	
		{down = device_commands.Button_1, up = device_commands.Button_1, cockpit_device_id  = devices.CONTROL_SYSTEM, value_down = 1.0, value_up = 0.6, name = _('Flaps Up else Takeoff (2-way Switch)'), category = {_('Custom'), _('Flight Control')}},
		{down = device_commands.Button_1, up = device_commands.Button_1, cockpit_device_id  = devices.CONTROL_SYSTEM, value_down = 0.0, value_up = 0.6, name = _('Flaps Landing else Takeoff (2-way Switch)'), category = {_('Custom'), _('Flight Control')}},

		-- Right Switch Panel

		{down = device_commands.Button_9, up = device_commands.Button_9, cockpit_device_id  = devices.RIGHT_SWITCH_PANEL, value_down = 1.0, value_up = 0.0, name = _('Position light tail BRIGHT else Off (2-way Switch)'), category ={_('Custom'), _('Right Switch Panel')}},
		{down = device_commands.Button_9, up = device_commands.Button_9, cockpit_device_id  = devices.RIGHT_SWITCH_PANEL, value_down = -1.0, value_up = 0.0, name = _('Position light tail DIM else Off (2-way Switch)'), category ={_('Custom'), _('Right Switch Panel')}},
		{down = device_commands.Button_8, up = device_commands.Button_8, cockpit_device_id  = devices.RIGHT_SWITCH_PANEL, value_down = 1.0, value_up = 0.0, name = _('Position light wing BRIGHT else Off (2-way Switch)'), category ={_('Custom'), _('Right Switch Panel')}},
		{down = device_commands.Button_8, up = device_commands.Button_8, cockpit_device_id  = devices.RIGHT_SWITCH_PANEL, value_down = -1.0, value_up = 0.0, name = _('Position light wing DIM else Off (2-way Switch)'), category ={_('Custom'), _('Right Switch Panel')}},

		-- Radiator Air Controls

		{down = device_commands.Button_1, up = device_commands.Button_1, cockpit_device_id = devices.ENGINE_SYSTEM, value_down = 0.3, value_up = 0.2 , name = _('Radiator coolant Open else Auto (2-way Switch)'), category = {_('Custom'), _('Radiator Air Controls')}},
		{down = device_commands.Button_1, up = device_commands.Button_1, cockpit_device_id = devices.ENGINE_SYSTEM, value_down = 0.1, value_up = 0.2 , name = _('Radiator coolant Close else Auto (2-way Switch)'), category = {_('Custom'), _('Radiator Air Controls')}},
		{down = device_commands.Button_2, up = device_commands.Button_2, cockpit_device_id = devices.ENGINE_SYSTEM, value_down = 0.3, value_up = 0.2 , name = _('Oil Control Open else Auto (2-way Switch)'), category = {_('Custom'), _('Radiator Air Controls')}},
		{down = device_commands.Button_2, up = device_commands.Button_2, cockpit_device_id = devices.ENGINE_SYSTEM, value_down = 0.1, value_up = 0.2 , name = _('Oil Control Close else Auto (2-way Switch)'), category = {_('Custom'), _('Radiator Air Controls')}},

**Comment by correca** [comment]
Note for TF-51:

In order for the proposed flap commands to work on the TF-51, it is first necessary to fix a bug, reported more than a year ago, in the file
"DCS World OpenBeta root game installation"\Mods\aircraft\TF-51D\Cockpit\Scripts\clickabledata.lua

With a text editor open the file clickabledata.lua and search for the string "Flaps control Handle". In the found line of code replace device_commands.Button_2 by device_commands.Button_1 and save te file.

The corrected line of code would look like this:
elements["pnt_94"] = multiposition_switch_limited(_("Flaps Control Handle"),devices.CONTROL_SYSTEM, device_commands.Button_1, 94, 6, 0.2)

You must make this correction manually or use a mod manager to apply this change after each new DCS update.

**Comment by Munkwolf** [comment]
Thanks! Those commands are now in there. I also did a refresh like I did with the Sabre and added a bunch of other stuff. Canopy emergency release handle on right and the bomb release levers on the left work surprisingly well as axis commands.

The flaps commands worked ok for me on the TF-51 without modifying the clickabledata.

Also added the radiator/oil open/close else auto commands.. those seem a little strange to me? Seems you're basically either max open else auto or max close else auto? I also added an auto-else-manual option for those.

Next up is P-47.

**Comment by correca** [comment]
I use radiator/oil commands with 3-way Warthog switches so that in the center position it is automatic and open and close at the ends.The idea is to preferably use the automatic position and only open or close the radiators to quickly correct overtemperatures or undertemperatures and then return to the automatic position.

**Comment by correca** [comment]
The bug within clickabledata of the TF-51 exists. With the default file, the flaps do not move correctly following the position of the yellow flap lever in the cockpit.

With default clickcabledata file, the flaps are adjusted to full flaps with the yellow lever but the flaps move only 10 degrees.
![imagen](https://user-images.githubusercontent.com/100794967/179422589-e131e896-e177-4e34-888e-9ffb2175c700.png)

With the file correction, the flap response is correct as you move the yellow lever.
![imagen](https://user-images.githubusercontent.com/100794967/179422706-c3197bb1-7371-47fc-b8ad-5a5192eb4aa9.png)

**Reply by Munkwolf** [reply]
Ah, got it. I had pilot body on and didn't notice the yellow lever not positioning correctly.
